### ISSUE TITLE
Ord compare

### ISSUE THREAD

**Issue by goodmind** [rpominov/static-land#49]
what about `Ord`'s `compare` function? 
```ts
compare(a: A, b: A): -1 | 0 | 1
```

**Comment by davidchambers** [comment]
We can have [`sort`][1] without the `-1`/`0`/`1` ugliness required by `Array#sort`. ;)


[1]: https://sanctuary.js.org/#sort

**Comment by goodmind** [comment]
@davidchambers but you can derive `compare` from `lte`

**Comment by davidchambers** [comment]
What's the advantage of `compare` over `lte`?

**Comment by goodmind** [comment]
Sanctuary also doesn't seem to support static-land

**Comment by davidchambers** [comment]
> Sanctuary also doesn't seem to support static-land

I'm not intimately familiar with Static Land, so please correct me if I am wrong, but my understanding is that the Sanctuary module is a polymorphic Static Land module.

We can create values of type `Maybe a` or `Either a b` by using the `of` function of the `S` module:

```javascript
> S.of (S.Maybe) (64)
Just (64)

> S.of (S.Either) (64)
Right (64)
```

We can interact with these values by using the `map` function of the `S` module:

```javascript
> S.map (Math.sqrt) (S.of (S.Maybe) (64))
Just (8)

> S.map (Math.sqrt) (S.of (S.Either) (64))
Right (8)
```

If this does not count as Static Land support, I'd love to know what more Sanctuary could provide.

**Comment by goodmind** [comment]
@davidchambers if I have static land `Ord` module, how do I use sanctuary sort?

**Comment by davidchambers** [comment]
Good question, @goodmind! I see the problem now. @rpominov, perhaps you could help with this. How would one change `sort :: (Ord a, Applicative m, Foldable m, Monoid (m a)) => m a -> m a` to work with Static Land? Must it take `TypeRep m` and `TypeRep a` as additional arguments?

```diff
-S.sort (['foo', 'bar', 'baz'])
+S.sort (Array) (String) (['foo', 'bar', 'baz'])
```

**Comment by rpominov** [comment]
Right! Since #45 isn't resolved yet, the only way to implement `sort` is to pass modules/TypeReps for both types explicitly.

**Comment by davidchambers** [comment]
Thanks, Roman. #45 will be terrific, in that case. :D
